app.js

```diff
@@ -108,81 +119,168 @@ app.get('/auth/google', passport.authenticate('google', {
 
-app.post('/search', function (req, res) {
-    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('search'))
-        console.log("error getting data");
-    storageModule.findOrg(req.body.search.location).then(function (data) {
-        res.send(data);
+app.post('/search', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('searchStr'))
+        res.send("Error getting data. Please try again later");
+    storageModule.fetchUser(req.body.searchStr).then(function (data) {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(data));
     }).otherwise(function (err) {
-        res.send("No search results");
-    })
-})
+        res.end("No search results");
+    });
+});
+
+app.get('/otherProfile', isAuthenticated, function (req, res) {
+    res.sendFile(__dirname + "/views/html/" + "otherProfile.html");
+});
+
+app.post('/otherProfile', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('othPro'))
+        res.send("Error.Please try again later");
+    else {
+        console.log(req.body.othPro);
+        res.redirect("/ticket");
+    }
+});
+
+app.get('/ticket', isAuthenticated, function (req, res) {
+    res.sendFile(__dirname + "/views/html/" + "ticket.html");
+});
+
+app.post('/ticket', isAuthenticated, function (req, res) {
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+    if (!req.hasOwnProperty('body') || !req.hasOwnProperty('user') || req.user.length <= 0)
+        res.end(JSON.stringify("Error Please try again later"));
+    else {
+        storageModule.createTicket(req.body.citName, req.body.citEmail, req.body.orgName, req.body.orgEmail, req.body.startDate, req.body.endDate, req.body.TDescr, req.body.type).then(function (data) {
+            res.end(JSON.stringify(data));
+        }).otherwise(function (err) {
+            res.end(JSON.stringify("Error Please try again later"));
+        });
+    }
+});
 
-app.get('/editProfile', function (req, res) {
+app.get('/editProfile', isAuthenticated, function (req, res) {
     res.sendFile(__dirname + "/views/html/" + "editProfile.html");
 });
 
-app.post('/editProfile', function (req, res) {
-    if (!req.hasOwnProperty('body') || !req.body.hasOwnProperty('name'))
-        console.log("error getting data");
-    storageModule.updateUser(json.stringify(req.body)).then(function (data) {
-        console.log("Succesfully edited!")
+app.post('/editProfile', isAuthenticated, function (req, res) {
+    if (!req.hasOwnProperty('body') || !req.hasOwnProperty('user') || req.user.length <= 0)
+        res.send("Error.Please try again later");
+    storageModule.updateUser(req.body, req.user[0]).then(function (data) {
+        res.send("Successfully edited");
     }).otherwise(function (err) {
-        console.log("Error updating! Please try again later!")
+        res.send("Error updating in. Please try again later!");
```

Comment:
see doing res.send - by default sends a 200 status code which in httprequest terms means a success so for this we  need to explictly respond with a status code for error.
Here as it is a server side error and there is status code for that 
https://www.restapitutorial.com/httpstatuscodes.html
If you refer to this 500 is code for the Internal server error
so all in all it should be 
res.status(500).send........